File: in1000/oblig8/celle.py
class Celle:
    # Konstruktør
    def __init__(self):
        self._status = "doed"
        self._naboer = []
        self._ant_levende_naboer = 0

    def sett_doed(self): #setter status av cellen til død
        self._status = "doed"

    def sett_levende(self): #setter status av cellen til levende
        self._status = "levende"

    def legg_til_nabo(self, nabo): #legger til nabocelle i liste av naboer
        return self._naboer.append(nabo)

    def er_levende(self): #returnerer cellens status om den er levende (True) eller ikke (False)
        return self._status == "levende"

    def hent_status(self):
        pass

    def hent_status_tegn(self): #metode som representerer celle status med tegn
        if self.er_levende():
            return "O" #levende celle representert med O
        return "." #død celle representert med .

    def tell_levende_naboer(self):
        antall_levende = [] #ny liste skal fylles med levende celler
        for celle in self._naboer: #itererer gjennom liste av naboer
            if celle.er_levende(): #hvis celle er levende
                antall_levende.append(celle) #legger til levende celle i liste

        self._ant_levende_naboer = len(antall_levende) #ant levende naboer er da lengden av listen

        #kunne gjort dette, men virker litt fy-fy:
        """
        self._ant_levende_naboer = 0
        for antall_levende in self._naboer:
            if antall_levende.er_levende() == True:
                self._ant_levende_naboer += 1

        return self._ant_levende_naboer
        """
    #metode består av nøstet if-setning som hopper til else hvis antall naboer
    #verken er mindre enn 2 eller større enn 3
    def oppdater_status(self):
        if self.er_levende():
            if self._ant_levende_naboer > 3:
                self.sett_doed()
            elif self._ant_levende_naboer < 2:
                self.sett_doed()
        else: #hvis cellen er dau i utgangspunket
            if self._ant_levende_naboer == 3: #cellen vil leve i neste generasjon
                self.sett_levende()             #hvis den har 3 naboceller

File: IN1010/2023/oblig_6/ERLEND_GUI/Celle.java


public class Celle{

    private int antNaboer;
    private int antLevendeNaboer;
    private boolean levende;
    private Celle[] naboer;

    public Celle() {
        antNaboer = 0;
        antLevendeNaboer = 0;
        levende = false;
        naboer = new Celle[8];
    }


    public boolean erLevende() {
        return levende;
    }

    public void settDoed() {
        levende = false;
    }

    public void settLevende() {
        levende = true;
    }

    public char hentStatusTegn() {  // returnerer String "O" eller".
        if (levende == true){
            return 'O';
        }
        else {
            return '.';
        }
    }

    public void leggTilNabo(Celle nabo) {   // looper gjennom liste med naboer: hvis element er null: sett nabo på den indeksen.
        boolean b = true;

        for (int i = 0; i < naboer.length; i++){
            if (naboer[i] == null && b == true) { 
                naboer[i] = nabo;
                b = false;
                antNaboer++;
            }
        }
    }

    public void tellLevendeNaboer() {   // looper gjennom liste med naboer: ++ for hver nabo hvor .levende == true.
        antLevendeNaboer = 0;
        
        for (int i = 0; i < naboer.length; i++) {
            if (naboer[i] != null){
                if (naboer[i].erLevende()) {
                    antLevendeNaboer++;
                }
            }
        }
    }


    public void oppdaterStatus() {   // ulike statements som bestemmer om celle.levende settes true / false.

        if (erLevende()) {
            if (antLevendeNaboer < 2) {
                settDoed();
            }
            else if (antLevendeNaboer > 3) {
                settDoed();
            }
        }

        else {
            if (antLevendeNaboer == 3) {
                settLevende();
            }
        }
    }
}